src/main/java/seedu/address/storage/JsonAdaptedTeachingAssistant.java

```diff
@@ -96,7 +98,18 @@ public Person toModelType() throws IllegalValueException {
         final ModuleCode modelModuleCode = new ModuleCode(getModuleCode());
 
         final Set<Tag> modelTags = new HashSet<>(personTags);
-        return new TeachingAssistant(modelName, modelModuleCode, modelPhone, modelEmail, modelGender, modelTags);
+
+        if (getLocation() == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, Location.class.getSimpleName()));
+        }
```

Comment:
Similar to comment above

Reply:
I agree location could be an optional field. For this iteration, I made location a compulsory field just like the other fields (name, email etc.).